modules/javafx.base/src/main/java/javafx/collections/ModifiableObservableListBase.java

```diff
@@ -101,28 +101,46 @@ public boolean setAll(Collection<? extends E> col) {
 
     @Override
     public boolean addAll(Collection<? extends E> c) {
+        if (c.isEmpty()) {
+            return false;
+        }
+
         beginChange();
         try {
-            boolean res = super.addAll(c);
-            return res;
+            return super.addAll(c);
         } finally {
             endChange();
         }
     }
 
     @Override
     public boolean addAll(int index, Collection<? extends E> c) {
+        if (c.isEmpty()) {
+            if (index < 0 || index > size()) {
+                throw new IndexOutOfBoundsException("Index: " + index);
+            }
+
+            return false;
+        }
+
```

Comment:
Shouldn't this always check the index?

```suggestion
        if (index < 0 || index > size()) {
            throw new IndexOutOfBoundsException("Index: " + index);
        }
        if (c.isEmpty()) {  // implicit null check
            return false;
        }

```